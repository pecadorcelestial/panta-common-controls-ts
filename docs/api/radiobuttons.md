# RadioButton

### Properties

| Name    | Type            | Required | Comments |
| --------|-----------------|----------|----------|
| checked | bool            | false    | Sets the initial state of the **radiobutton**. |
| value   | string / number | true     | Sets the value to be returned when the component is checked. |

### Functions

| Name     | Parameters | Type | Description |
|----------|------------|------|-------------|
| onChange | checked    | bool | Returns the value of the **radiobutton**. |

### Methods

| Name     | Parameters | Type | Description |
|----------|------------|------|-------------|
| getValue | N/A        | N/A  | Returns the state of the **checkbox**. |
| setValue | value      | bool | Changes the value of the **checkbox**. |

## Usage

```javascript
import React, { Component } from 'react';
import { RadioButton } from 'panda-common-controls';

class BasicForm extends Component {
    render() {
        //Properties.
        const props = {
            //Required.
            value: 1,
            //Optional.
            checked: false,
            //Functions.
            onChange: (value) => {console.log('[panda-common-controls][test][onChange] Valor: ', value);}
        };
        //Result.
        return(
            <div>
                <RadioButton {...props}>Option #1</RadioButton>
            <div>
        );
    }
}
```

# RadioButtonsGroup

### Properties

| Name          | Type            | Required | Comments |
| --------------|-----------------|----------|----------|
| options       | object          | true     | Object that will contain ``RadioButtons`` to be _gropued_. [1] |
| selectedValue | string / number | true     | Sets the value to be returned when the component is checked. |

[1]
```json
[
    {
        "name": "Option #1",
        "value": 1
    },
    {
        "name": "Option #2",
        "value": 2
    }
]
```

### Functions

| Name     | Parameters | Type | Description |
|----------|------------|------|-------------|
| onChange | checked    | bool | Returns the selected value. |

### Methods

| Name     | Parameters | Type            | Description |
|----------|------------|-----------------|-------------|
| getValue | N/A        | N/A             | Returns the selected value of the **group**. |
| setValue | value      | string / number | Changes the selected value of the **group**. |

## Usage

```javascript
import React, { Component } from 'react';
import { RadioButtonsGroup } from 'panda-common-controls';

class BasicForm extends Component {
    render() {
        //Options.
        const options = [
            {
                "name": "Option #1",
                "value": 1
            },
            {
                "name": "Option #2",
                "value": 2
            }
        ];
        //Properties.
        const props = {
            //Required.
            options,
            selectedValue: 1,
            //Functions.
            onChange: (value) => {console.log('[panda-common-controls][test][onChange] Valor: ', value);}
        };
        //Result.
        return(
            <div>
                <RadioButtonsGroup {...props}/>
            <div>
        );
    }
}
```
