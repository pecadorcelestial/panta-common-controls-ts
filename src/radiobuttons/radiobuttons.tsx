//Componentes generales.
import React, { Component, HTMLProps } from 'react';
import PropTypes from 'prop-types';
import styled, { AnyStyledComponent } from 'styled-components';

const Input: AnyStyledComponent<HTMLProps<HTMLLabelElement>, any> = styled.label`
    box-sizing: border-box;
    color: #242424;
    cursor: pointer;
    display: inline-block;
	height: 30px;
	font-family: "Open Sans", sans-serif;
	font-size: 16px;
	font-weight: normal;
	font-style: normal;
	font-stretch: normal;
	letter-spacing: normal;
	margin: 0px;
	padding: 5px 0px 0px 25px;
	position: relative;
	text-align: left;
	width: auto;
    white-space: nowrap;
`;

interface ICheckMarkProps {
    checked: boolean;
};
const CheckMark: AnyStyledComponent<ICheckMarkProps & HTMLProps<HTMLDivElement>, any> = styled.div`
    background-color: ${(props: ICheckMarkProps) => props.checked ? `#1476FB`: `#BFBFBF`};
    border: none;
    border-radius: 50%;
    cursor: pointer;
    height: 20px;
    left: 0;
    margin: 0px;
    position: absolute;
    top: 5px;
    width: 20px;
    
	transition: all .3s;
	
    ${Input}:hover & {
        background-color: ${(props: ICheckMarkProps) => props.checked ? `#11AAFF`: `#A7A7A7`};
    }

    ${(props: ICheckMarkProps) => props.checked ? `
        &:after {
            border: solid white;
            border-width: 0px 3px 3px 0px;
            content: "";
            display: block;
            height: 10px;
            left: 6px;
            position: absolute;
            top: 2px;
            width: 5px;

            transform: rotate(45deg);
        }` : ``}
`;

interface IRadioButtonProps extends ICheckMarkProps {
    value: string | number;
    onChange?: Function;
};
export class RadioButton extends Component<IRadioButtonProps, {}> {
    render() {
        return(
            <Input {...this.props} onClick={() => { if(this.props.onChange) { this.props.onChange(this.props.value); } }}>
                <CheckMark checked={this.props.checked}/>
                {this.props.children}
            </Input>
        );
    }
}

// GGGG RRRR  U   U PPPP   OOO
//G     R   R U   U P   P O   O
//G  GG RRRR  U   U PPPP  O   O
//G   G R   R U   U P     O   O
// GGGG R   R  UUU  P      OOO

const Layout: AnyStyledComponent<HTMLProps<HTMLDivElement>, any> = styled.div`
    box-sizing: border-box;
    height: auto;
    margin: 0px;
    padding: 0px;
    width: auto;
`;

const Buttons: AnyStyledComponent<HTMLProps<HTMLDivElement>, any> = styled.div`
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    height: auto;
    margin: 0px;
    padding: 0px;
    width: 100%;
`;

interface IOption {
    name: string;
    value: string | number;
};
interface IRadioButtonsGroupProps {
    options: Array<IOption>;
    selectedValue: string | number;
    onChange?: Function;
};
interface IRadioButtonsGroupState {
    selectedValue: string | number;
};
export class RadioButtonsGroup extends Component<IRadioButtonsGroupProps, IRadioButtonsGroupState> {
    //*** CONSTRUCTOR ***
    constructor(props: IRadioButtonsGroupProps) {
        super(props);
        this.state = {
            selectedValue: this.props.selectedValue
        };
    }
    //*** MÃ‰TODOS ***
    getValue = (): string | number => {
        return this.state.selectedValue;
    }
    setValue = (value: string | number): void => {
        this.setState({ selectedValue: value });
    }
    //*** RESULTADO ***
    render() {
        return(
            <Layout>
                <Buttons>
                    {
                        this.props.options.map((option, index) => <RadioButton key={`rbt-opt-${index}`} value={option.value} checked={option.value == this.state.selectedValue} onChange={(value: string | number) => { this.setState({ selectedValue: value }); if(this.props.onChange) { this.props.onChange(value); } }}>{option.name}</RadioButton>)
                    }
                </Buttons>
            </Layout>
        );
    }
}