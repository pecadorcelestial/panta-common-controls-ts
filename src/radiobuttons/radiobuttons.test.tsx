
//RRRR  EEEEE  AAA   CCCC TTTTT
//R   R E     A   A C       T
//RRRR  EEE   AAAAA C       T
//R   R E     A   A C       T
//R   R EEEEE A   A  CCCC   T

import React from 'react';
import renderer from 'react-test-renderer';

//EEEEE N   N ZZZZZ Y   Y M   M EEEEE
//E     NN  N    Z   Y Y  MM MM E
//EEE   N N N   Z     Y   M M M EEE
//E     N  NN  Z      Y   M   M E
//EEEEE N   N ZZZZZ  YYY  M   M EEEEE

import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });

//JJJJJ EEEEE  SSSS TTTTT
//    J E     S       T
//    J EEE    SSS    T
//J   J E         S   T
// JJJ  EEEEE SSSS    T

//NOTA: Éste módulo ayuda a que los errores en componentes estilizados se vean de manera correcta:
//https://medium.com/styled-components/effective-testing-for-styled-components-67982eb7d42b
import 'jest-enzyme';
import 'jest-styled-components';

// CCCC  OOO  M   M PPPP   OOO  N   N EEEEE N   N TTTTT EEEEE  SSSS
//C     O   O MM MM P   P O   O NN  N E     NN  N   T   E     S
//C     O   O M M M PPPP  O   O N N N EEE   N N N   T   EEE    SSS
//C     O   O M   M P     O   O N  NN E     N  NN   T   E         S
// CCCC  OOO  M   M P      OOO  N   N EEEEE N   N   T   EEEEE SSSS

import { RadioButton, RadioButtonsGroup } from './radiobuttons';

//Snapshots.
describe('[SNAPSHOT][Componentes][Common][RadioButton] - RadioButton normal.', () => {
	//UNCHECKED.
	it('Debe pintar el componente con los siguientes modificadores: (Checked: false).', () => {
		//Funciones 'dummy'.
		const handleOnChange = event => {
			//Nada que hacer aquí.
        };
		//Propiedades.		
        const radiobuttonProps = {
            //Obligatorios.
            value: 1,
            //Opcionales.
            checked: false,
            //Funciones.
            onChange: handleOnChange
        };
        //Se crea el componente.
        
        //U   U N   N  CCCC H   H EEEEE  CCCC K   K EEEEE DDDD
        //U   U NN  N C     H   H E     C     K  K  E     D   D
        //U   U N N N C     HHHHH EEE   C     KKK   EEE   D   D
        //U   U N  NN C     H   H E     C     K  K  E     D   D
        // UUU  N   N  CCCC H   H EEEEE  CCCC K   K EEEEE DDDD

        //Snapshot.
        const component = renderer.create(<RadioButton {...radiobuttonProps}>Opción #1</RadioButton>).toJSON();
		expect(component).toMatchSnapshot();
    });
    //CHECKED.
	it('Debe pintar el componente con los siguientes modificadores: (Checked: false).', () => {
		//Funciones 'dummy'.
		const handleOnChange = event => {
			//Nada que hacer aquí.
        };
		//Propiedades.		
        const radiobuttonProps = {
            //Obligatorios.
            value: 1,
            //Opcionales.
            checked: true,
            //Funciones.
            onChange: handleOnChange
        };
        //Se crea el componente.
        
        // CCCC H   H EEEEE  CCCC K   K EEEEE DDDD
        //C     H   H E     C     K  K  E     D   D
        //C     HHHHH EEE   C     KKK   EEE   D   D
        //C     H   H E     C     K  K  E     D   D
        // CCCC H   H EEEEE  CCCC K   K EEEEE DDDD

        //Snapshot.
        const component = renderer.create(<RadioButton {...radiobuttonProps}>Opción #1</RadioButton>).toJSON();
		expect(component).toMatchSnapshot();
    });
});

//Flujo #1 (Con función onChange existente).
describe('[FLUJO][Componentes][Common][RadioButton] - Valida que la función "onChange" sea llamada.', () => {
    //Funciones 'dummy'.
    const handleOnChange = jest.fn(event => {
		//Nada que hacer aquí.
	});
    //Propiedades.		
    const radiobuttonProps = {
        //Obligatorios.
        value: 1,
        //Opcionales.
        checked: false,
        //Funciones.
        onChange: handleOnChange
    };
	//Se crea el componente.
	//NOTA: Al utilizar la función 'mount' se detona las siguientes funciones: constructor, componentDidMount y render.
    const component = enzyme.mount(<RadioButton {...radiobuttonProps}>Opción #1</RadioButton>);

	// OOO  N   N  CCCC L     IIIII  CCCC K   K
	//O   O NN  N C     L       I   C     K  K
	//O   O N N N C     L       I   C     KKK
	//O   O N  NN C     L       I   C     K  K
	// OOO  N   N  CCCC LLLLL IIIII  CCCC K   K

	it('Debe mandar llamar la función "onChange" que se pasa como propiedad.', () => {
		//Simulación.
		component.simulate('click');
		//Expectativa.
        expect(handleOnChange).toHaveBeenCalled();
	});
});

//Flujo #2 (Con función onChange inexistente).
describe('[FLUJO][Componentes][Common][RadioButton] - Valida que la función "onChange" NO sea llamada.', () => {
    //Funciones 'dummy'.
    const handleOnChange = jest.fn(event => {
		//Nada que hacer aquí.
	});
    //Propiedades.		
    const radiobuttonProps = {
        //Obligatorios.
        value: 1,
        //Opcionales.
        checked: false
    };
	//Se crea el componente.
	//NOTA: Al utilizar la función 'mount' se detona las siguientes funciones: constructor, componentDidMount y render.
    const component = enzyme.mount(<RadioButton {...radiobuttonProps}>Opción #1</RadioButton>);

	// OOO  N   N  CCCC L     IIIII  CCCC K   K
	//O   O NN  N C     L       I   C     K  K
	//O   O N N N C     L       I   C     KKK
	//O   O N  NN C     L       I   C     K  K
	// OOO  N   N  CCCC LLLLL IIIII  CCCC K   K

	it('Debe mandar llamar la función "onChange" que se pasa como propiedad.', () => {
		//Simulación.
		component.simulate('click');
		//Expectativa.
        expect(handleOnChange).not.toHaveBeenCalled();
	});
});

//Snapshots.
describe('[SNAPSHOT][Componentes][Common][RadioButtonsGroup] - RadioButtonsGroup normal.', () => {
	//UNCHECKED.
	it('Debe pintar el componente con un grupo de componentes tipo "RadioButton".', () => {
		//Funciones 'dummy'.
		const handleOnChange = event => {
			//Nada que hacer aquí.
        };
        //Opcions.        
        const radioButtons = [
            {
                name: 'Option #1',
                value: 1
            },
            {
                name: 'Option #2',
                value: 2
            },
            {
                name: 'Option #3',
                value: 3
            }
        ];
		//Propiedades.		
        const radiobuttonsGroupProps = {
            //Obligatorios.
            options: radioButtons,
            //Opcionales.
            selectedValue: 2,
            //Funciones.
            onChange: handleOnChange
        };
        //Se crea el componente.
        //Snapshot.
        const component = renderer.create(<RadioButtonsGroup {...radiobuttonsGroupProps}/>).toJSON();
		expect(component).toMatchSnapshot();
    });
});

//Flujo #1 + Métodos.
describe('[FLUJO][Componentes][Common][RadioButtonsGroup] - Valida el cambio de estado al dar clic sobre el componente.', () => {
    //Funciones 'dummy'.
    const handleOnChange = jest.fn(event => {
		//Nada que hacer aquí.
	});
    //Opcions.        
    const radioButtons = [
        {
            name: 'Option #1',
            value: 1
        },
        {
            name: 'Option #2',
            value: 2
        },
        {
            name: 'Option #3',
            value: 3
        }
    ];
    //Propiedades.		
    const radiobuttonsGroupProps = {
        //Obligatorios.
        options: radioButtons,
        //Opcionales.
        selectedValue: 2,
        //Funciones.
        onChange: handleOnChange
    };
	//Se crea el componente.
	//NOTA: Al utilizar la función 'mount' se detona las siguientes funciones: constructor, componentDidMount y render.
    const component = enzyme.mount(<RadioButtonsGroup {...radiobuttonsGroupProps}/>);
    //NOTA: La función "find" devuelve la cantidad de elementos del tipo seleccionado, aunque por alguna razón no se pueden tratar como un arreglo.
    let radioButtonArray = component.find('RadioButton');
    //NOTA: Por otro lado, esta función también devuelve el componente completo, por lo que devuelve también todos los componentes dentro:
    /*
        ┌──────────────────────────────────────────────────┐
        │ RADIOBUTTON                      key='rbt-opt-0' │
        │    ┌───────────────────────────────────────────┐ │
        │    │ INPUT                                     │ │
        │    │   ┌─────────────────────────────────────┐ │ │
        │    │   │ DIV                                 │ │ │
        │    │   │   ┌───────────────────────────────┐ │ │ │
        │    │   │   │ {CHILDREN}                    │ │ │ │
        │    │   │   └───────────────────────────────┘ │ │ │
        │    │   └─────────────────────────────────────┘ │ │
        │    └───────────────────────────────────────────┘ │
        └──────────────────────────────────────────────────┘
    */
    //Es decir que por cada búsqueda, suponiendo que se hiciera por la propiedad key, el resultado serían 4 componentes por cada wrapper y sólo 1 contendría
    //dicha propiedad.
    let radioButton = radioButtonArray.findWhere(node => node.key() && node.key() === 'rbt-opt-0');
    it('Se verifica que se hayan pintado 3 componentes tipo "RadioButton".', () => {
        expect(radioButtonArray).toHaveLength(3);
        expect(radioButton).toHaveLength(1);
    });

	// CCCC  OOO  M   M PPPP   OOO  N   N EEEEE N   N TTTTT DDDD  IIIII DDDD  M   M  OOO  U   U N   N TTTTT
	//C     O   O MM MM P   P O   O NN  N E     NN  N   T   D   D   I   D   D MM MM O   O U   U NN  N   T
	//C     O   O M M M PPPP  O   O N N N EEE   N N N   T   D   D   I   D   D M M M O   O U   U N N N   T
	//C     O   O M   M P     O   O N  NN E     N  NN   T   D   D   I   D   D M   M O   O U   U N  NN   T
	// CCCC  OOO  M   M P      OOO  N   N EEEEE N   N   T   DDDD  IIIII DDDD  M   M  OOO   UUU  N   N   T

	it('Se verifica que la información inicial esté correcta.', () => {
		//Expectativas.
        expect(component.state().selectedValue).toBe(2);
	});

	// OOO  N   N  CCCC L     IIIII  CCCC K   K
	//O   O NN  N C     L       I   C     K  K
	//O   O N N N C     L       I   C     KKK
	//O   O N  NN C     L       I   C     K  K
	// OOO  N   N  CCCC LLLLL IIIII  CCCC K   K
    
	it('No debe mandar llamar la función "onChange" que se pasa como propiedad, y cambiar el valor "checked" a true.', () => {
		//Simulación.
        radioButton.prop('onChange')(1);
		//Expectativa.
        expect(handleOnChange).toHaveBeenCalled();
        expect(component.state().selectedValue).toBe(1);
    });
    
    //V   V  AAA  L      OOO  RRRR
    //V   V A   A L     O   O R   R
    //V   V AAAAA L     O   O RRRR
    // V V  A   A L     O   O R   R
    //  V   A   A LLLLL  OOO  R   R
    
	it('Debe asignar el valor enviado en el control.', () => {
		//Simulación.
		component.instance().setValue(3);
		//Expectativa.
		expect(component.state().selectedValue).toBe(3);
    });
    it('Debe obtener el valor asignado al control.', () => {
        //Simulación.
        let value = component.instance().getValue();
        //Expectativa.
        expect(value).toBe(3);
    });
});

//Flujo #2 (Con función onChange inexistente).
describe('[FLUJO][Componentes][Common][RadioButtonsGroup] - Valida el cambio de estado al dar clic sobre el componente.', () => {
    //Funciones 'dummy'.
    const handleOnChange = jest.fn(event => {
		//Nada que hacer aquí.
	});
    //Opcions.        
    const radioButtons = [
        {
            name: 'Option #1',
            value: 1
        },
        {
            name: 'Option #2',
            value: 2
        },
        {
            name: 'Option #3',
            value: 3
        }
    ];
    //Propiedades.		
    const radiobuttonsGroupProps = {
        //Obligatorios.
        options: radioButtons,
        //Opcionales.
        selectedValue: 2,
        //Funciones.
        //onChange: handleOnChange
    };
	//Se crea el componente.
	//NOTA: Al utilizar la función 'mount' se detona las siguientes funciones: constructor, componentDidMount y render.
    const component = enzyme.mount(<RadioButtonsGroup {...radiobuttonsGroupProps}/>);
    //NOTA: La función "find" devuelve la cantidad de elementos del tipo seleccionado, aunque por alguna razón no se pueden tratar como un arreglo.
    let radioButtonArray = component.find('RadioButton');
    //NOTA: Por otro lado, esta función también devuelve el componente completo, por lo que devuelve también todos los componentes dentro:
    /*
        ┌──────────────────────────────────────────────────┐
        │ RADIOBUTTON                      key='rbt-opt-0' │
        │    ┌───────────────────────────────────────────┐ │
        │    │ INPUT                                     │ │
        │    │   ┌─────────────────────────────────────┐ │ │
        │    │   │ DIV                                 │ │ │
        │    │   │   ┌───────────────────────────────┐ │ │ │
        │    │   │   │ {CHILDREN}                    │ │ │ │
        │    │   │   └───────────────────────────────┘ │ │ │
        │    │   └─────────────────────────────────────┘ │ │
        │    └───────────────────────────────────────────┘ │
        └──────────────────────────────────────────────────┘
    */
    //Es decir que por cada búsqueda, suponiendo que se hiciera por la propiedad key, el resultado serían 4 componentes por cada wrapper y sólo 1 contendría
    //dicha propiedad.
    let radioButton = radioButtonArray.findWhere(node => node.key() && node.key() === 'rbt-opt-0');
    it('Se verifica que se hayan pintado 3 componentes tipo "RadioButton".', () => {
        expect(radioButtonArray).toHaveLength(3);
        expect(radioButton).toHaveLength(1);
    });

	// CCCC  OOO  M   M PPPP   OOO  N   N EEEEE N   N TTTTT DDDD  IIIII DDDD  M   M  OOO  U   U N   N TTTTT
	//C     O   O MM MM P   P O   O NN  N E     NN  N   T   D   D   I   D   D MM MM O   O U   U NN  N   T
	//C     O   O M M M PPPP  O   O N N N EEE   N N N   T   D   D   I   D   D M M M O   O U   U N N N   T
	//C     O   O M   M P     O   O N  NN E     N  NN   T   D   D   I   D   D M   M O   O U   U N  NN   T
	// CCCC  OOO  M   M P      OOO  N   N EEEEE N   N   T   DDDD  IIIII DDDD  M   M  OOO   UUU  N   N   T

	it('Se verifica que la información inicial esté correcta.', () => {
		//Expectativas.
        expect(component.state().selectedValue).toBe(2);
	});

	// OOO  N   N  CCCC L     IIIII  CCCC K   K
	//O   O NN  N C     L       I   C     K  K
	//O   O N N N C     L       I   C     KKK
	//O   O N  NN C     L       I   C     K  K
	// OOO  N   N  CCCC LLLLL IIIII  CCCC K   K
    
	it('No debe mandar llamar la función "onChange" que se pasa como propiedad, y cambiar el valor "checked" a true.', () => {
		//Simulación.
        radioButton.prop('onChange')(1);
		//Expectativa.
        expect(handleOnChange).not.toHaveBeenCalled();
        expect(component.state().selectedValue).toBe(1);
    });
});
