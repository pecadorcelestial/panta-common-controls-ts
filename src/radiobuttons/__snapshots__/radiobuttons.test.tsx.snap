// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SNAPSHOT][Componentes][Common][RadioButton] - RadioButton normal. Debe pintar el componente con los siguientes modificadores: (Checked: false). 1`] = `
.c1 {
  box-sizing: border-box;
  color: #242424;
  cursor: pointer;
  display: inline-block;
  height: 30px;
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0px;
  padding: 5px 0px 0px 25px;
  position: relative;
  text-align: left;
  width: auto;
  white-space: nowrap;
}

.c2 {
  background-color: #BFBFBF;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  height: 20px;
  left: 0;
  margin: 0px;
  position: absolute;
  top: 5px;
  width: 20px;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c0:hover .c2 {
  background-color: #A7A7A7;
}

.c0:hover .c3 {
  background-color: #11AAFF;
}

<label
  checked={false}
  className="c0 c1"
  onChange={[Function]}
  onClick={[Function]}
  value={1}
>
  <div
    checked={false}
    className="c2"
  />
  Opción #1
</label>
`;

exports[`[SNAPSHOT][Componentes][Common][RadioButton] - RadioButton normal. Debe pintar el componente con los siguientes modificadores: (Checked: false). 2`] = `
.c1 {
  box-sizing: border-box;
  color: #242424;
  cursor: pointer;
  display: inline-block;
  height: 30px;
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0px;
  padding: 5px 0px 0px 25px;
  position: relative;
  text-align: left;
  width: auto;
  white-space: nowrap;
}

.c0:hover .c3 {
  background-color: #A7A7A7;
}

.c2 {
  background-color: #1476FB;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  height: 20px;
  left: 0;
  margin: 0px;
  position: absolute;
  top: 5px;
  width: 20px;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c0:hover .c2 {
  background-color: #11AAFF;
}

.c2:after {
  border: solid white;
  border-width: 0px 3px 3px 0px;
  content: "";
  display: block;
  height: 10px;
  left: 6px;
  position: absolute;
  top: 2px;
  width: 5px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<label
  checked={true}
  className="c0 c1"
  onChange={[Function]}
  onClick={[Function]}
  value={1}
>
  <div
    checked={true}
    className="c2"
  />
  Opción #1
</label>
`;

exports[`[SNAPSHOT][Componentes][Common][RadioButtonsGroup] - RadioButtonsGroup normal. Debe pintar el componente con un grupo de componentes tipo "RadioButton". 1`] = `
.c3 {
  box-sizing: border-box;
  color: #242424;
  cursor: pointer;
  display: inline-block;
  height: 30px;
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0px;
  padding: 5px 0px 0px 25px;
  position: relative;
  text-align: left;
  width: auto;
  white-space: nowrap;
}

.c4 {
  background-color: #BFBFBF;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  height: 20px;
  left: 0;
  margin: 0px;
  position: absolute;
  top: 5px;
  width: 20px;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c2:hover .c4 {
  background-color: #A7A7A7;
}

.c5 {
  background-color: #1476FB;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  height: 20px;
  left: 0;
  margin: 0px;
  position: absolute;
  top: 5px;
  width: 20px;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c2:hover .c5 {
  background-color: #11AAFF;
}

.c5:after {
  border: solid white;
  border-width: 0px 3px 3px 0px;
  content: "";
  display: block;
  height: 10px;
  left: 6px;
  position: absolute;
  top: 2px;
  width: 5px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c0 {
  box-sizing: border-box;
  height: auto;
  margin: 0px;
  padding: 0px;
  width: auto;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: auto;
  margin: 0px;
  padding: 0px;
  width: 100%;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      checked={false}
      className="c2 c3"
      onChange={[Function]}
      onClick={[Function]}
      value={1}
    >
      <div
        checked={false}
        className="c4"
      />
      Option #1
    </label>
    <label
      checked={true}
      className="c2 c3"
      onChange={[Function]}
      onClick={[Function]}
      value={2}
    >
      <div
        checked={true}
        className="c5"
      />
      Option #2
    </label>
    <label
      checked={false}
      className="c2 c3"
      onChange={[Function]}
      onClick={[Function]}
      value={3}
    >
      <div
        checked={false}
        className="c4"
      />
      Option #3
    </label>
  </div>
</div>
`;
